internal/services/cognitive/cognitive_account_data_source.go

```diff
@@ -100,32 +258,69 @@ func dataSourceCognitiveAccountRead(d *pluginsdk.ResourceData, meta interface{})
 	if model := resp.Model; model != nil {
 		d.Set("location", location.NormalizeNilable(model.Location))
 		d.Set("kind", model.Kind)
+
 		if sku := model.Sku; sku != nil {
 			d.Set("sku_name", sku.Name)
 		}
 
 		if props := model.Properties; props != nil {
 			if apiProps := props.ApiProperties; apiProps != nil {
-				d.Set("qna_runtime_endpoint", apiProps.QnaRuntimeEndpoint)
+				d.Set("custom_question_answering_search_service_id", pointer.From(apiProps.QnaAzureSearchEndpointId))
+				d.Set("metrics_advisor_aad_client_id", pointer.From(apiProps.AadClientId))
+				d.Set("metrics_advisor_aad_tenant_id", pointer.From(apiProps.AadTenantId))
+				d.Set("metrics_advisor_super_user_name", pointer.From(apiProps.SuperUser))
+				d.Set("metrics_advisor_website_name", pointer.From(apiProps.WebsiteName))
+				d.Set("qna_runtime_endpoint", pointer.From(apiProps.QnaRuntimeEndpoint))
 			}
-			d.Set("endpoint", props.Endpoint)
 
-			localAuthEnabled := true
-			if props.DisableLocalAuth != nil {
-				localAuthEnabled = !*props.DisableLocalAuth
+			d.Set("custom_subdomain_name", pointer.From(props.CustomSubDomainName))
+
+			customerManagedKey, err := flattenCognitiveAccountCustomerManagedKey(props.Encryption)
```

Comment:
could we create a separate function here specifically for flattening the customerManagedKey for the datasource?